src/ray/common/grpc_util.h

```diff
@@ -87,19 +87,31 @@ inline grpc::Status RayStatusToGrpcStatus(const Status &ray_status) {
   }
 }
 
+inline std::string GrpcStatusToRayStatusMessage(const grpc::Status &grpc_status) {
+  return absl::StrCat("RPC Error message: ", grpc_status.error_message(),
+                      "; RPC Error details: ", grpc_status.error_details());
+}
+
 /// Helper function that converts a gRPC status to ray status.
 inline Status GrpcStatusToRayStatus(const grpc::Status &grpc_status) {
   if (grpc_status.ok()) {
     return Status::OK();
   } else if (grpc_status.error_code() == grpc::StatusCode::DEADLINE_EXCEEDED) {
     // DEADLINE_EXCEEDED means the gRPC request has timedout. Convert it to Ray timeout
     // status code.
-    return Status(StatusCode::TimedOut,
-                  absl::StrCat("RPC Error message: ", grpc_status.error_message(),
-                               " RPC Error details: ", grpc_status.error_details()));
-  } else {
+    return Status(StatusCode::TimedOut, GrpcStatusToRayStatusMessage(grpc_status));
+  } else if (grpc_status.error_code() == grpc::StatusCode::ABORTED) {
+    // This is a status generated by ray code.
+    // See RayStatusToGrpcStatus for details.
     return Status(Status::StringToCode(grpc_status.error_message()),
                   grpc_status.error_details());
+  } else if (grpc_status.error_code() == grpc::StatusCode::UNAVAILABLE) {
+    return Status::GrpcUnavailable(GrpcStatusToRayStatusMessage(grpc_status));
+  } else {
+    // TODO(jjyao) Use GrpcUnknown as the catch-all status for all
```

Comment:
Is it intentional this is left as it is? I remember we will only handle gRPC unavailable, and put everything else as just IOError?